soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -4463,38 +3729,99 @@ void DrawRandoEditor(bool& open) {
+                        for (auto locationIt : RandomizerCheckObjects::GetAllRCObjects()) {
+                            if (!excludedLocations.count(locationIt.second.rc) &&
+                                locationIt.second.vOrMQ != RCVORMQ_MQ && //disable all MQ checks until we support them
+                                locationIt.second.rcType != RCTYPE_SHOP && //disable shops until we have shopsanity
+                                locationIt.second.rcType != RCTYPE_GOSSIP_STONE &&
+                                locationIt.second.rcType != RCTYPE_LINKS_POCKET &&
+                                locationIt.second.rcType != RCTYPE_CHEST_GAME &&
+                                ((locationIt.second.rcType != RCTYPE_SKULL_TOKEN) ||
+                                 (CVar_GetS32("gRandomizeShuffleTokens", 0) == 3) || // all tokens
+                                 ((CVar_GetS32("gRandomizeShuffleTokens", 0) == 2) && RandomizerCheckObjects::AreaIsOverworld(areaIt.first)) || // overworld tokens
+                                 ((CVar_GetS32("gRandomizeShuffleTokens", 0) == 1) && RandomizerCheckObjects::AreaIsDungeon(areaIt.first)) // dungeon tokens
+                                ) &&
+                                locationIt.second.rcArea == areaIt.first &&
+                                locationSearch.PassFilter(locationIt.second.rcSpoilerName.c_str())) {
+
+                                hasItems = true;
+                                break;
+                            }
+                        }
+
+                        if (hasItems) {
+                            ImGui::SetNextItemOpen(true, ImGuiCond_Once);
+                            if (ImGui::TreeNode(areaIt.second.c_str())) {
+                                for (auto locationIt : RandomizerCheckObjects::GetAllRCObjects()) {
+                                    if (!excludedLocations.count(locationIt.second.rc) &&
+                                        locationIt.second.vOrMQ != RCVORMQ_MQ && //disable all MQ checks until we support them
+                                        locationIt.second.rcType != RCTYPE_SHOP && //disable shops until we have shopsanity
+                                        locationIt.second.rcType != RCTYPE_GOSSIP_STONE &&
+                                        locationIt.second.rcType != RCTYPE_LINKS_POCKET &&
+                                        locationIt.second.rcType != RCTYPE_CHEST_GAME &&
+                                        ((locationIt.second.rcType != RCTYPE_SKULL_TOKEN) ||
+                                        (CVar_GetS32("gRandomizeShuffleTokens", 0) == 3) || // all tokens
+                                        ((CVar_GetS32("gRandomizeShuffleTokens", 0) == 2) && RandomizerCheckObjects::AreaIsOverworld(areaIt.first)) || // overworld tokens
+                                        ((CVar_GetS32("gRandomizeShuffleTokens", 0) == 1) && RandomizerCheckObjects::AreaIsDungeon(areaIt.first)) // dungeon tokens
```

Comment:
Maybe it would be better to have `GetAllRCObjects`  handle this logic? Or a different appropriately named function?

Reply:
great call. filtering it before iterating will be much better than adding more checks in here

Reply:
@leggettc18 i decided to try out adding a `visibleInImgui` bool to the `RandomizerCheckObject` struct. i then made it so those values are updated once when we switch to the locations tab (instead of running the logic every frame)

i think i like how this is working now but i'm very open to suggestions https://github.com/HarbourMasters/Shipwright/pull/1249/commits/7fdb194dadb080fbe6fed95e9ccb1d88ef119889

Reply:
I like it! You probably want a `#define` or a helper function or something for constructing the list entries. It would've saved you from adding `false` to all 700 list entries lol. I'd still go back and do it so that when potsanity becomes a thing and we need to add several hundred more checks it'll save us some time.

Reply:
https://github.com/HarbourMasters/Shipwright/pull/1249/commits/fb3320c3ed1a15927c2e0cd95624731469f6763c